hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -707,29 +929,75 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
                 )
 
         if sell_fill_quantity > 0:
+            # Maker sell
+            # Taker buy
+            taker_slippage_adjustment_factor = Decimal("1") + self._slippage_buffer
+
+            if self.is_gateway_market(market_pair.taker):
+                taker_price = await market_pair.taker.market.get_order_price(
+                    taker_trading_pair,
+                    True,
+                    sell_fill_quantity * base_rate
+                )
+            else:
+                taker_price = taker_market.get_price_for_volume(
+                    taker_trading_pair,
+                    True,
+                    sell_fill_quantity * base_rate
+                ).result_price
+
             hedged_order_quantity = min(
-                sell_fill_quantity,
-                taker_market.c_get_available_balance(market_pair.taker.quote_asset) /
-                market_pair.taker.get_price_for_volume(True, sell_fill_quantity).result_price *
-                self._order_size_taker_balance_factor
+                sell_fill_quantity * base_rate,
+                taker_market.get_available_balance(market_pair.taker.quote_asset) /
+                taker_price * self._order_size_taker_balance_factor
             )
-            quantized_hedge_amount = taker_market.c_quantize_order_amount(taker_trading_pair, Decimal(hedged_order_quantity))
-            taker_top = taker_market.c_get_price(taker_trading_pair, True)
+            quantized_hedge_amount = taker_market.quantize_order_amount(
+                taker_trading_pair,
+                Decimal(hedged_order_quantity)
+            )
+
             avg_fill_price = (sum([r.price * r.amount for _, r in sell_fill_records]) /
                               sum([r.amount for _, r in sell_fill_records]))
-            order_price = taker_market.get_price_for_volume(taker_trading_pair, True,
-                                                            quantized_hedge_amount).result_price
+
+            maker_order_ids = [r.order_id for _, r in sell_fill_records]
+            maker_exchange_trade_ids = [r.exchange_trade_id for _, r in sell_fill_records]
+
+            if len(maker_order_ids) != 1 or len(maker_exchange_trade_ids) != 1:
+                self.logger().error("Multiple sell maker orders fills")
+                raise RuntimeError
```

Comment:
Comment in `if` stmt applies here as well.

Reply:
I have the same comment here as in the previous issue.

Reply:
This can also be  marked as resolved when you wish since it's not a critical issue.